enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+When the plain provisioner's BundleDeployment controller sees a BundleDeployment resource that contains a desired Bundle template, and that template doesn't match any Bundles that exist on the cluster, a new Bundle will be generated to match that desired configuration. In this case, the plain provisioner will generate a Bundle that gets sourced by the registry provisioner, and wait until that generated Bundle reports an unpacked phase.
+
+The registry provisioner has logic that converts OLM’s registry+v1 bundle format into a plain bundle format. The OLM ClusterServiceVersion (CSV) resource can be decomposed into a set of core Kubernetes manifests (e.g. RBAC, Deployment, etc.). This conversion logic allows the registry provisioner to effectively act as a bridge between higher-level components (e.g. PO manager) that want to apply registry+v1 bundle contents, and still have the rukpak stack manage those contents.
+
+After the registry provisioner successfully unpacks the registry+v1 bundle, and converts that into a plain bundle format, the plain provisioner will attempt to apply those manifests. In the case that an individual manifest is invalid, and the Kubernetes API server rejects the creation of that manifest, then the plain provisioner will update the BundleDeployment resource to reflect that failed rollout.
+
+> Note: The plain provisioner has the current limitation around applying Bundle contents that will produce runtime issues.
+
+In the case that unpacked Bundle contents contain a Deployment manifest, and that manifest references an invalid container image, then the result would be an unavailable container process after being applied to the cluster. Despite the presence of runtime issues here, the BundleDeployment that manages those manifests would reflect a successful installation state. Addressing this limitation in the context of PO packages is discussed in more detail in subsequent sections.
+
+#### Proxying PO State
+
+```mermaid
+sequenceDiagram
+participant CVO
+participant ClusterOperator
+participant PO Manager
+CVO->>ClusterOperator: Watches
+PO Manager->>ClusterOperator: Manages
+PO Manager-->ClusterOperator: Aggregates PO Package State
+PO Manager->>PO Custom Resources: Watches
+PO Custom Resources->>PO Manager: Relays .Status
+BundleDeployments->>PO Custom Resources: Relays .Status
+```
+
+The PO manager will be responsible for acting as a proxy between the CVO, and the underlying PO packages during phase 0. This can be done by reconciling an "aggregate" ClusterOperator that reflects the overall state of installing PO packages. Future phases may require PO packages to reconcile OLM's OperatorCondition API, or the directly manage a ClusterOperator resource to express their health and overall state.
```

Comment:
> or the directly manage a ClusterOperator resource to express their health and overall state.

they'll probably never do this one, because the CVO ignores COs that it didn't create/reconcile.

but i could see OLM introducing a new api to replace operatorcondition, and expecting operators to manage that to report on their status.

Reply:
Is there an opportunity to change this behavior at the CVO level to support dynamically generating ClusterOperator resources, or would a first-class API be beneficial long term here? It feels like the local UX around `kubectl get co` is a nice interface to the cluster components, and having POs be listed in that output feels reasonable to me.

Reply:
I'll add this as an open question that we can track over time given we won't have any runway during 4.12 to explore alternatives in this domain.